core/src/main/java/com/linecorp/armeria/server/RoutingTrie.java

```diff
@@ -223,16 +223,25 @@ private Node<V> checkNode(Node<V> node, String path, int begin, boolean exact, I
                 break;
             case PARAMETER:
                 // Consume characters until the delimiter '/' as a path variable.
-                final int delim = path.indexOf('/', begin);
-                if (delim < 0) {
-                    // No more delimiter.
-                    return node;
+                final int delimSlash = path.indexOf('/', begin);
+
+                if (delimSlash < 0) {
+                    final int lastColon = path.lastIndexOf(':');
+                    // lastIndexOf performs backward search
+                    if (lastColon < begin) {
+                        // No more delimiter.
+                        return node;
+                    } else {
+                        final Node<V> verb = node.children.get(':');
```

Comment:
I guess `verb.path` could differ from the string after the last `:`.

For example, `"/hello/foo:no-verb"` should be matched with `noVerb()` method.
```java
class MyService {
    @Get("/hello/{foo}:get")
    public String get() {
        return "Hello, world!:get";
    }

    @Get("/hello/{foo}")
    public String noVerb() {
        return "Hello, world!:no-verb";
    }
}
```